src/Microsoft.ML.Data/Scorers/ScoreMapperSchema.cs

```diff
@@ -1,330 +0,0 @@
-        {
-        }
-
-        public override ColumnType GetColumnType(int col)
-        {
-            Contracts.CheckParam(0 <= col && col < ColumnCount, nameof(col));
-            if (col == base.ColumnCount)
-                return NumberType.Float;
-            return base.GetColumnType(col);
-        }
-
-        public override bool TryGetColumnIndex(string name, out int col)
-        {
-            Contracts.CheckValue(name, nameof(name));
-            if (name == MetadataUtils.Const.ScoreValueKind.Probability)
-            {
-                col = base.ColumnCount;
-                return true;
-            }
-            return base.TryGetColumnIndex(name, out col);
-        }
-
-        public override string GetColumnName(int col)
-        {
-            Contracts.CheckParam(0 <= col && col < ColumnCount, nameof(col));
-            if (col == base.ColumnCount)
-                return MetadataUtils.Const.ScoreValueKind.Probability;
-            return base.GetColumnName(col);
-        }
-
-        public override IEnumerable<KeyValuePair<string, ColumnType>> GetMetadataTypes(int col)
-        {
-            Contracts.CheckParam(0 <= col && col < ColumnCount, nameof(col));
-            var items = base.GetMetadataTypes(col);
-            if (col == base.ColumnCount)
-                items = items.Prepend(BoolType.Instance.GetPair(MetadataUtils.Kinds.IsNormalized));
-            return items;
-        }
-
-        public override ColumnType GetMetadataTypeOrNull(string kind, int col)
-        {
-            Contracts.CheckNonEmpty(kind, nameof(kind));
-            Contracts.CheckParam(0 <= col && col < ColumnCount, nameof(col));
-
-            if (col == base.ColumnCount && kind == MetadataUtils.Kinds.IsNormalized)
-                return BoolType.Instance;
-            return base.GetMetadataTypeOrNull(kind, col);
-        }
-
-        public override void GetMetadata<TValue>(string kind, int col, ref TValue value)
-        {
-            Contracts.CheckNonEmpty(kind, nameof(kind));
-            Contracts.CheckParam(0 <= col && col < ColumnCount, nameof(col));
-
-            if (col == base.ColumnCount && kind == MetadataUtils.Kinds.IsNormalized)
-                MetadataUtils.Marshal<bool, TValue>(IsNormalized, col, ref value);
-            else
-                base.GetMetadata<TValue>(kind, col, ref value);
-        }
-
-        private void IsNormalized(int col, ref bool dst)
-        {
-            dst = true;
-        }
-
-        protected override void GetScoreValueKind(int col, ref ReadOnlyMemory<char> dst)
-        {
-            Contracts.Assert(0 <= col && col < ColumnCount);
-            if (col == base.ColumnCount)
-                dst = MetadataUtils.Const.ScoreValueKind.Probability.AsMemory();
-            else
-                base.GetScoreValueKind(col, ref dst);
-        }
-    }
-
-    public sealed class SequencePredictorSchema : ScoreMapperSchemaBase
```

Comment:
@Ivanidzo4ka - that's the intention.  `ISchema` is going away, so any code (internal or public) will need to be updated.

Reply:
@ivmatan@microsoft.com, after running 5 out of 7 sequence-related TLC tests, I still don't find a place where this object is actually used. In TLC code base, SequencePredictorSchema's constructor always bypasses the part which distinguish it from ScoreMapperSchemaBase because `keyNames` is always 0. Therefore, we don't even know if this class was created correctly. Is a replacement to this class is required?

---
In reply to: [246872132](https://github.com/dotnet/machinelearning/pull/2107#discussion_r246872132) [](ancestors = 246872132)

Reply:
Just added it back with tests. Please take a look.

---
In reply to: [246871022](https://github.com/dotnet/machinelearning/pull/2107#discussion_r246871022) [](ancestors = 246871022)